### PR TITLE
Cleanup deprecated configuration flags in agent

### PR DESCRIPTION
**Pull request by howardjohn**
These options have been deprecated and replaced with just configuring
mesh config directly. For most users, this change has no impact. This
only will impact people directly setting these flags (eg not using our
Helm templates), which is not really something we ever claim any degree
of support for.

For https://github.com/istio/istio/issues/21222

### CODE REVIEW

**Comment by howardjohn** [comment]
This is blocked by https://github.com/istio/istio/pull/21428/files because I don't want to do things twice

**Review by hzxuzhonghu — APPROVED** [review]

**Review by elfinhe — APPROVED** [review]

File: tools/packaging/common/istio-start.sh [hunk 1]
```diff
@@ -103,19 +100,19 @@ ISTIO_AGENT_FLAGS=${ISTIO_AGENT_FLAGS:-}
 # Split ISTIO_AGENT_FLAGS by spaces.
 IFS=' ' read -r -a ISTIO_AGENT_FLAGS_ARRAY <<< "$ISTIO_AGENT_FLAGS"
 
+export MESH_CONFIG=${MESH_CONFIG:-"
+defaultConfig:
+  serviceCluster: $SVC
+  controlPlaneAuthPolicy: ${CONTROL_PLANE_AUTH_POLICY}
```
Comment (rshriram) [inline tools/packaging/common/istio-start.sh]: why cant we make this mandatory? with istiod,

**Review by rshriram — COMMENTED** [review]
mostly fine. lets get rid of control plane auth completely. its not accurate in the istiod model as we are doing one-way tls.

File: tools/packaging/common/istio-start.sh (hunk 1 above)
Comment (howardjohn) [reply tools/packaging/common/istio-start.sh]: Blocked by https://github.com/istio/istio/pull/21658 I believe. I can followup

**Review by douglas-reid — APPROVED** [review]
approving for p&t related changes.

**Review by nmittler — APPROVED** [review]

**Pull request merged**